bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineManagerImpl.java

```diff
@@ -51,6 +54,9 @@
 @NonNullByDefault
 @Component(service = ScriptEngineManager.class)
 public class ScriptEngineManagerImpl implements ScriptEngineManager {
+    private static final String THREAD_POOL_NAME = "OH-scriptenginemanager";
+
+    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(THREAD_POOL_NAME);
```

Comment:
If it only occassionaly runs these tasks is might be better to reuse `ThreadPoolManager.THREAD_POOL_NAME_COMMON` for this instead of creating a new thread pool. 

```java
    /**
     * The common thread pool is reserved for occasional, light weight tasks that run quickly, and
     * use little resources to execute. Tasks that do not fit into this category should setup
     * their own dedicated pool or permanent thread.
     */
    public static final String THREAD_POOL_NAME_COMMON = "common";
```

Does it only schedule these tasks when rules are deleted/modified? Do you know how resource heavy the scheduled tasks will be?

Reply:
I decided for the new thread pool to rule out that it is the same thread as the calling one. But I believe the `ScriptFileWatcher` has its own thread pool, correct?

Yes, only on delete/modify. I'm not sure how heavy the tasks are, probably not too much. And it currently only affects Jython, so I'll change it. We can change it in the future if we experience issues.